### DISCUSSION TITLE
File-based repo settings

### DISCUSSION THREAD

**Discussion by qwerty287** [woodpecker-ci/woodpecker#2317] [General]
Woodpecker currently has two main locations for configuration: The `.woodpecker.yaml` file and the repo settings (and if you use the configuration service which shouldn't be covered by this discussion).

My idea is to provide a way to configure most of the settings you can change in the repo settings using properties in `.woodpecker.yaml` or some other file. Of course this shouldn't take the settings from PRs changing them, so only sync them on pushes to the default branch, and also should not cover secrets and registries.
This mainly has the advantage that you only have one configuration location which is also file-based.

This is not a security issue from side of viewing settings because you can see the full settings even if you're logged out using the API.
It is one though because everybody with push access could push changes and so gains admin rights. This can be changed though that settings changes are only applied and synced if the commit is pushed by someone with admin rights.

There's a similar thing for Github with [this 3rd-party app](https://github.com/apps/settings).

Related: https://github.com/woodpecker-ci/woodpecker/issues/2312#issuecomment-1687288844 https://github.com/woodpecker-ci/woodpecker/discussions/2283

**Comment by lonix1** [comment]
Related: https://github.com/woodpecker-ci/woodpecker/discussions/2328
